core/network/fabric-interop-cc/interfaces/asset-mgmt/asset_locks.go

```diff
@@ -0,0 +1,828 @@
+/*
+ * Copyright IBM Corp. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package main
+
+import (
+    "fmt"
+    "errors"
+    "time"
+    "strconv"
+    "encoding/json"
+
+    "github.com/golang/protobuf/proto"
+    "github.com/hyperledger/fabric-chaincode-go/shim"
+    log "github.com/sirupsen/logrus"
+    "github.com/hyperledger-labs/weaver-dlt-interoperability/core/network/fabric-interop-cc/interfaces/asset-mgmt/protos-go/common"
+)
+
+
+const(
+    defaultTimeLockMillis = 5 * 60 * 1000      // 5 minutes
+)
+
+
+// AssetManagement provides functions to manage asset locks by calling suitable functions in the interop contract
+type AssetManagement struct {
+    shim.Chaincode
+    interopChaincodeId string
+}
+
+
+// Utility functions
+func (am *AssetManagement) Configure(interopChaincodeId string) {
+    am.interopChaincodeId = interopChaincodeId
+}
+
+
+// Ledger transaction (invocation) functions
+
+func (am *AssetManagement) AddFungibleAssetCount(stub shim.ChaincodeStubInterface, assetType string, numUnits int) (bool, error) {
```

Comment:
it is less likely this will be a notable issue but it might be safer to replace the use of `int`, whose size (32 or 64) is platform-dependent (which could cause non-determinism), with something more fixed .. ideally `int64`, to avoid overflows

Reply:
I think `int` is pretty much guaranteed to be 32 bits, isn't it? Most languages other than Go use `long` for 64-bit integers. I can change this to `int32` to make it explicit. But why do you think it should be `int64`? 32 bits already allow us to express numbers in excess of 2 trillion.

Reply:
- The size of `int` in Go is platform-dependent, and not guaranteed to be 32 bit: https://golangbyexample.com/go-size-range-int-uint/
- I think 32 bits would be up to 2 billion? not trillion

Reply:
also, should we use unsigned int type instead?

Reply:
Right, it's over 2 billion. Still sufficiently large for a fungible asset count, no?

I think using `int` and flagging an error when a negative value is passed in OK. Why do you prefer `uint`?

Reply:
The reason I think 2 Billion would be small for fungible tokens is that often what we are tracking with this number is the smallest possible denomination of a fungible asset (since this is an int and not a floating-point number). For instance, for Bitcoin that would be _Satoshi_, and for Ethereum that would be _Wei_. 2 Billion would not be enough to represent 22BTC (2,200,000,000 Satoshi) or even 1 ETH (1,000,000,000,000,000,000 wei)

Reply:
regarding `int` vs `uint`: my thinking here is why dynamically enforce through code what we can statically enforce through the type system? ... but I don't really mind either way in this context

Reply:
Done. Changed everything to `uint64`.